### ISSUE TITLE
`Err` value: ParseFloatError

### ISSUE THREAD

**Issue by lspk** [Xandaros/ic10lsp#33]
Hi,

When using the last version of this language server with @awilliamson plugin I got this error:

```rust
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseFloatError { kind: Invalid }', src/main.rs:1303:59
stack backtrace:
   0: rust_begin_unwind
             at /rustc/8460ca823e8367a30dda430efda790588b8c84d3/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/8460ca823e8367a30dda430efda790588b8c84d3/library/core/src/panicking.rs:64:14
   2: core::result::unwrap_failed
             at /rustc/8460ca823e8367a30dda430efda790588b8c84d3/library/core/src/result.rs:1790:5
   3: ic10lsp::Backend::run_diagnostics::{{closure}}
   4: <ic10lsp::Backend as tower_lsp::LanguageServer>::did_open::{{closure}}
   5: tower_lsp::jsonrpc::router::Router<S,E>::method::{{closure}}::{{closure}}::{{closure}}
   6: <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll
   7: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   8: <core::pin::Pin<P> as core::future::future::Future>::poll
   9: <core::pin::Pin<P> as core::future::future::Future>::poll
  10: <tower_lsp::service::LspService<S> as tower_service::Service<tower_lsp::jsonrpc::Request>>::call::{{closure}}
  11: <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll
  12: futures_util::stream::stream::StreamExt::poll_next_unpin
  13: <futures_util::stream::stream::buffer_unordered::BufferUnordered<St> as futures_core::stream::Stream>::poll_next
  14: <futures_util::stream::stream::filter_map::FilterMap<St,Fut,F> as futures_core::stream::Stream>::poll_next
  15: <futures_util::stream::stream::forward::Forward<St,Si,Item> as core::future::future::Future>::poll
  16: <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll
  17: <futures_util::future::poll_fn::PollFn<F> as core::future::future::Future>::poll
  18: tower_lsp::transport::Server<I,O,L>::serve::{{closure}}
  19: tokio::runtime::park::CachedParkThread::block_on
  20: tokio::runtime::runtime::Runtime::block_on
  21: ic10lsp::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by Xandaros** [comment]
Can you share the file you opened that lead to this error? Or did this happen on a blank file?

**Comment by lspk** [comment]
> Can you share the file you opened that lead to this error? Or did this happen on a blank file?

```asm
s db Setting -1
define volumePump -454028979
define liquidPipeValve -517628750
define GasSensor -1252983604
define PipeAnalyzer -2113838091
define StationBattery -1388288459
define GasGenerator 1165997963
define StirlingEngine -260316435
define GasMixer 2104106366
define TankSmallInsulated 272136332
define pumpCooling HASH("ACCoolingPump")
define coolingValve HASH("ACCoolingValve")
define ACGasSensor HASH("ACGasSensor")
define ACPipeAnalyzer HASH("ACPipeAnalyzer")
define FuelMixer HASH("FuelMixer")
define FuelTank HASH("FuelTank")
define MaxTemperature 300 # enable valve
define MinTemperatureWater 283
define MaxFuelPressure 5
define LargeBatteryMin 7000000
define LargeBatteryMax 8800000
# 0 idle no water external
# 1 cooling, valve enabled pump disabled
# 2 post-cooling, valve disabled, pump enabled
alias state r8
sbn PipeAnalyzer ACPipeAnalyzer On 1
yield
lbn r0 PipeAnalyzer ACPipeAnalyzer VolumeOfLiquid 2
move state 0
bnez r0 emptypipes
sbn PipeAnalyzer ACPipeAnalyzer On 0
move state 0
stateidle:
yield
s db Setting state
bne state 0 statecooling
lbn r0 GasSensor ACGasSensor Temperature 2
ble r0 MaxTemperature statecooling
move state 1
sbn liquidPipeValve coolingValve On 1
sbn volumePump pumpCooling On 0
sbn PipeAnalyzer ACPipeAnalyzer On 1
ble r0 323 statecooling
sb GasGenerator On 0 # too hot stop gens
sb StirlingEngine On 0
yield
statecooling:
bne state 1 statepostcooling
lbn r0 PipeAnalyzer ACPipeAnalyzer Temperature 2
bgt r0 MinTemperatureWater statepostcooling
emptypipes:
sbn liquidPipeValve coolingValve On 0
sbn volumePump pumpCooling On 1
move state 2
statepostcooling:
bne state 2 endstate
lbn r0 PipeAnalyzer ACPipeAnalyzer VolumeOfLiquid 2
bnez r0 endstate
sbn liquidPipeValve coolingValve On 0
sbn volumePump pumpCooling On 0
sbn PipeAnalyzer ACPipeAnalyzer On 0
move state 0
endstate:
lbn r1 TankSmallInsulated FuelTank Pressure 2
slt r15 r1 MaxFuelPressure
sbn GasMixer FuelMixer On r15
lb r0 StationBattery Charge 2
slt r15 r0 LargeBatteryMin
beqz r15 if3exit
lbn r0 GasSensor ACGasSensor Temperature 2
bgt r0 323 if3exit
sb StirlingEngine On 1
ble r0 279 if3exit
sb GasGenerator On 1
if3exit:
sgt r15 r0 LargeBatteryMax
beqz r15 if4exit
sb GasGenerator On 0
sb StirlingEngine On 0
if4exit:

j stateidle
```

thanks

**Comment by Xandaros** [comment]
Fixed in a04e45a02caaa9e81b09f2705d20164904e71a20
